app/src/main/java/com/concordium/wallet/data/walletconnect/Payload.kt

```diff
@@ -1,9 +1,19 @@
 package com.concordium.wallet.data.walletconnect
 
-data class Payload(
-    val address: ContractAddress,
-    val amount: String,
-    var maxEnergy: Long,
-    val message: String,
-    val receiveName: String
-)
+sealed interface Payload {
+
+    data class ComplexPayload(
```

Comment:
This complex payload is related to a specific transaction.

We should used naming which help understand the transaction is related to some type. This seems like a transaction to a contract, and the one below a transfer between accounts.

Reply:
changed to SimpleTransaction and UpdateTransaction